### PR TITLE
raftstore: reduce clones when compact regions

### PR DESCRIPTION
**Pull request by Hijiao**
Signed-off-by: Max <xiahaijiao@hotmail.com>

## What have you changed? (mandatory)

Reduce `cf.clone()`, `start_key.clone()` and `end_key.clone()` when compact range cf.

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
No

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

File: src/raftstore/store/worker/compact.rs [hunk 1]
```diff
@@ -106,17 +106,17 @@ impl Runner {
     /// Sends a compact range command to RocksDB to compact the range of the cf.
     pub fn compact_range_cf(
         &mut self,
-        cf_name: String,
-        start_key: Option<Vec<u8>>,
-        end_key: Option<Vec<u8>>,
+        cf_name: &str,
+        start_key: Option<&Vec<u8>>,
```
Comment (BusyJay) [inline src/raftstore/store/worker/compact.rs]: Take `Option<&[u8]>` instead.

File: src/raftstore/store/worker/compact.rs [hunk 2]
```diff
@@ -106,17 +106,17 @@ impl Runner {
...
+        end_key: Option<&Vec<u8>>,
     ) -> Result<(), Error> {
         let handle = box_try!(rocksdb::get_cf_handle(&self.engine, &cf_name));
```
Comment (BusyJay) [inline src/raftstore/store/worker/compact.rs]: Redundant `&`.

**Comment by Connor1996** [comment]
LGTM. Please fix the conflict

**Comment by siddontang** [comment]
```
test cases::test_stale_read::test_stale_read_during_merging ... thread 'cases::test_stale_read::test_stale_read_during_merging' panicked at 'find no region for "k1"', components/test_raftstore/src/cluster.rs:606:9
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:211
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:491
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:398
   6: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:353
   7: <test_raftstore::cluster::Cluster<T>>::get_region_with
             at ./<::std::macros::panic macros>:8
   8: failpoints::cases::test_stale_read::test_stale_read_during_merging
             at tests/failpoints/cases/test_stale_read.rs:312
   9: failpoints::cases::test_stale_read::test_stale_read_during_merging::{{closure}}
             at tests/failpoints/cases/test_stale_read.rs:226
  10: core::ops::function::FnOnce::call_once
             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/libcore/ops/function.rs:238
  11: <F as alloc::boxed::FnBox<A>>::call_box
             at src/libtest/lib.rs:1471
             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/libcore/ops/function.rs:238
             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/liballoc/boxed.rs:673
  12: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:102
```

PTAL @Connor1996

**Comment by Hijiao** [comment]
@zhangjinpeng1987  PTAL

**Comment by Hijiao** [comment]
@Connor1996  @siddontang  @zhangjinpeng1987 PTAL

**Pull request merged**